src/modules/location.rs

```diff
@@ -4,6 +4,35 @@ use reqwest::Client;
 use serde::Deserialize;
 
 use super::{config::Config, localization::Locales};
+use crate::modules::api::Api;
+
+struct OpenMeteoLocation<'a> {
+	address: &'a str,
+	language: &'a str,
+}
+
+impl<'a> Api for OpenMeteoLocation<'a> {
+	fn assemble(&self) -> String {
+		format!(
+			"https://geocoding-api.open-meteo.com/v1/search?name={}&language={}",
```

Comment:
Each supported API to be queried requires a fixed set of information.  I consider it useful to collect those in a struct and creating the query from it.  The data of the struct will be used to create the query URL for `reqwest`.  The URL template is specific to each API, so putting it hard-coded into the trait implementation prevents users from hard to find typos.